.meta/sources/socket.toml

```diff
@@ -15,17 +15,27 @@ null = false
 description = "The type of socket to use."
 
 [sources.socket.options.mode.enum]
-tcp = "The TCP protocol."
-udp = "The UDP protocol."
+tcp = "TCP Socket."
+udp = "UDP Socket."
+unix = "Unix Domain Socket."
 
 [sources.socket.options.address]
 type = "string"
 common = true
 examples = ["0.0.0.0:9000", "systemd", "systemd#3"]
 null = false
+relevant_when = {mode = ["tcp", "udp"]}
 description = """The address to listen for connections on, \
-or "systemd#N" to use the Nth socket passed by systemd socket activation. \
-Valid for "tcp" and "udp" sockets.\
+or "systemd#N" to use the Nth socket passed by systemd socket activation.\
+"""
+
+[sources.socket.options.path]
+type = "string"
+common = true
+examples = ["/path/to/socket"]
+null = false
+relevant_when = {mode = "unix"}
+description = """The unix socket path. *This should be absolute path*.\
```

Comment:
Should be or must be an absolute path? If it is a recommendation (should), I suggest adding a reason for the recommendation.

Reply:
This was copied verbatim from the equivalent config option in syslog.toml.
My guess as to why this restriction is there is because of the lack of something like the canonicalize() call I added in this CR.  But I'm actually thinking a change like that to add the canonicalization should be done and tested as its own issue if we want to do it, rather than snuck in here, so for now I'd like to leave this as-is.